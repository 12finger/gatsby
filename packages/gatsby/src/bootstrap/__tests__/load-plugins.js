const loadPlugins = require(`../load-plugins`)

describe(`Load plugins`, () => {
  it(`load plugins for a site`, async () => {
    let plugins = await loadPlugins({ plugins: [] })

    // Delete the resolve path as that varies depending
    // on platform so will cause snapshots to differ.
    plugins = plugins.map(plugin => {
      return {
        ...plugin,
        resolve: ``,
      }
    })

    expect(plugins).toMatchSnapshot()
  })

  it(`loads more complex plugins configuration`, async () => {
    const config = {
      plugins: [
        {
          resolve: `gatsby-source-filesystem`,
          options: {
            name: `pages`,
            path: `${__dirname}/src/pages/`,
          },
        },
        {
          resolve: `gatsby-transformer-remark`,
          options: {
            plugins: [
              {
                resolve: `gatsby-remark-responsive-image`,
                options: {
                  maxWidth: 690,
                },
              },
              {
                resolve: `gatsby-remark-responsive-iframe`,
              },
              `gatsby-remark-prismjs`,
              `gatsby-remark-copy-linked-files`,
              `gatsby-remark-smartypants`,
              `gatsby-remark-autolink-headers`,
            ],
          },
        },
        `gatsby-plugin-sharp`,
        {
          resolve: `gatsby-plugin-manifest`,
          options: {
            start_url: `/`,
            background_color: `#f7f7f7`,
            theme_color: `#191919`,
            display: `minimal-ui`,
          },
        },
        `gatsby-plugin-offline`,
      ],
    }

    let plugins = await loadPlugins(config)

    // Delete the resolve path as that varies depending
    // on platform so will cause snapshots to differ.
    plugins = plugins.map(plugin => {
      return {
        ...plugin,
        resolve: ``,
      }
    })

    expect(plugins).toMatchSnapshot()
  })
})
