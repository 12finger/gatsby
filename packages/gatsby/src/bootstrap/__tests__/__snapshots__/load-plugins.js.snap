// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Load plugins load plugins for a site 1`] = `
Array [
  Object {
    "name": "component-page-creator",
    "pluginOptions": Object {
      "plugins": Array [],
    },
    "resolve": "",
    "version": "1.0.0",
  },
  Object {
    "name": "internal-data-bridge",
    "pluginOptions": Object {
      "plugins": Array [],
    },
    "resolve": "",
    "version": "1.0.0",
  },
  Object {
    "name": "defaultSitePlugin",
    "pluginOptions": Object {
      "plugins": Array [],
    },
    "resolve": "",
    "version": "n/a",
  },
]
`;

exports[`Load plugins loads more complex plugins configuration 1`] = `
Array [
  Object {
    "name": "component-page-creator",
    "pluginOptions": Object {
      "plugins": Array [],
    },
    "resolve": "",
    "version": "1.0.0",
  },
  Object {
    "name": "internal-data-bridge",
    "pluginOptions": Object {
      "plugins": Array [],
    },
    "resolve": "",
    "version": "1.0.0",
  },
  Object {
    "name": "gatsby-source-filesystem",
    "pluginOptions": Object {
      "name": "pages",
      "path": "/Users/kylemathews/programs/gatsby/packages/gatsby/src/bootstrap/__tests__/src/pages/",
      "plugins": Array [],
    },
    "resolve": "",
    "version": "1.0.0-alpha15",
  },
  Object {
    "name": "gatsby-transformer-remark",
    "pluginOptions": Object {
      "plugins": Array [
        Object {
          "name": "gatsby-remark-responsive-image",
          "pluginOptions": Object {
            "maxWidth": 690,
            "plugins": Array [],
          },
          "resolve": "/Users/kylemathews/programs/gatsby/packages/gatsby-remark-responsive-image",
          "version": "1.0.0-alpha14",
        },
        Object {
          "name": "gatsby-remark-responsive-iframe",
          "pluginOptions": Object {
            "plugins": Array [],
          },
          "resolve": "/Users/kylemathews/programs/gatsby/packages/gatsby-remark-responsive-iframe",
          "version": "1.0.0-alpha14",
        },
        Object {
          "name": "gatsby-remark-prismjs",
          "pluginOptions": Object {
            "plugins": Array [],
          },
          "resolve": "/Users/kylemathews/programs/gatsby/packages/gatsby-remark-prismjs",
          "version": "1.0.0-alpha14",
        },
        Object {
          "name": "gatsby-remark-copy-linked-files",
          "pluginOptions": Object {
            "plugins": Array [],
          },
          "resolve": "/Users/kylemathews/programs/gatsby/packages/gatsby-remark-copy-linked-files",
          "version": "1.0.0-alpha15",
        },
        Object {
          "name": "gatsby-remark-smartypants",
          "pluginOptions": Object {
            "plugins": Array [],
          },
          "resolve": "/Users/kylemathews/programs/gatsby/packages/gatsby-remark-smartypants",
          "version": "1.0.0-alpha14",
        },
        Object {
          "name": "gatsby-remark-autolink-headers",
          "pluginOptions": Object {
            "plugins": Array [],
          },
          "resolve": "/Users/kylemathews/programs/gatsby/packages/gatsby-remark-autolink-headers",
          "version": "1.0.0-alpha14",
        },
      ],
    },
    "resolve": "",
    "version": "1.0.0-alpha15",
  },
  Object {
    "name": "gatsby-remark-responsive-image",
    "pluginOptions": Object {
      "maxWidth": 690,
      "plugins": Array [],
    },
    "resolve": "",
    "version": "1.0.0-alpha14",
  },
  Object {
    "name": "gatsby-remark-responsive-iframe",
    "pluginOptions": Object {
      "plugins": Array [],
    },
    "resolve": "",
    "version": "1.0.0-alpha14",
  },
  Object {
    "name": "gatsby-remark-prismjs",
    "pluginOptions": Object {
      "plugins": Array [],
    },
    "resolve": "",
    "version": "1.0.0-alpha14",
  },
  Object {
    "name": "gatsby-remark-copy-linked-files",
    "pluginOptions": Object {
      "plugins": Array [],
    },
    "resolve": "",
    "version": "1.0.0-alpha15",
  },
  Object {
    "name": "gatsby-remark-smartypants",
    "pluginOptions": Object {
      "plugins": Array [],
    },
    "resolve": "",
    "version": "1.0.0-alpha14",
  },
  Object {
    "name": "gatsby-remark-autolink-headers",
    "pluginOptions": Object {
      "plugins": Array [],
    },
    "resolve": "",
    "version": "1.0.0-alpha14",
  },
  Object {
    "name": "gatsby-plugin-sharp",
    "pluginOptions": Object {
      "plugins": Array [],
    },
    "resolve": "",
    "version": "1.0.0-alpha15",
  },
  Object {
    "name": "gatsby-plugin-manifest",
    "pluginOptions": Object {
      "background_color": "#f7f7f7",
      "display": "minimal-ui",
      "plugins": Array [],
      "start_url": "/",
      "theme_color": "#191919",
    },
    "resolve": "",
    "version": "1.0.0-alpha14",
  },
  Object {
    "name": "gatsby-plugin-offline",
    "pluginOptions": Object {
      "plugins": Array [],
    },
    "resolve": "",
    "version": "1.0.0-alpha14",
  },
  Object {
    "name": "defaultSitePlugin",
    "pluginOptions": Object {
      "plugins": Array [],
    },
    "resolve": "",
    "version": "n/a",
  },
]
`;
